patroni/dcs/consul.py

```diff
@@ -346,8 +346,7 @@ def touch_member(self, data, ttl=None, permanent=False):
             args = {} if permanent else {'acquire': self._session}
             self._client.kv.put(self.member_path, json.dumps(data, separators=(',', ':')), **args)
             if self._register_service:
-                self.update_service(self._my_member_data, data, force=create_member)
-            self._my_member_data = data
+                self.update_service(not create_member and member and member.data or {}, data)
```

Comment:
I had to resolve the conflict after the merge of https://github.com/zalando/patroni/pull/805, because it made `_my_member_data`  not available.
The `force` argument in the `update_service` is still available. I think it should be set to `True` during the next call if the last attempt to call `register_service` has failed.

Reply:
`update_service` now save last result and set `force` to `True` if last result was `False`
And because of this I've removed `retry` from `register_service`